.github/workflows/_ci.yaml

```diff
@@ -236,9 +286,10 @@ jobs:
       - name: Run integration test ${{ matrix.TEST_SCRIPT }}
         run: bash rosetta/tests/${{ matrix.TEST_SCRIPT }}
 
+  # If $TEST_SUBSET IN ("base", "jax")
   test-jax:
     needs: build-jax
-    if: inputs.ARCHITECTURE == 'amd64' # arm64 runners n/a
+    if: contains(fromJSON('["base", "jax"]'), inputs.TEST_SUBSET) && inputs.ARCHITECTURE == 'amd64' # arm64 runners n/a
```

Comment:
Is there a way to resolve this dependency from the job graph itself, since this is essentially the same information encoded in `needs`?

Reply:
I think it's fine to allow test-upstream-t5x to run if we're testing rosetta-t5x. Does that allow you to simplify the logic? 

It's still a big improvement if we're able to selectively trigger all t5x related nodes

Reply:
Unfortunately not, it was just one example but we have this issue within multiple slices of our deps graph. Whenever a node splits into several leaves, we have to limit which leaves should fire. `needs` just tells us that potentially all leaves could fire, but we need to narrow that down by our own selection.

I.e. now we build `jax` to be able to selectively test `t5x`, but don't want to run `test-jax`. The only way to really simplify it would be to get rid of `if: contains` but we can't do that since we have a tree. We can prune the set of triggers of course by grouping upstream/rosetta together, but so far I'm not creative enough to see a solution that doesn't require `contains`.

Reply:
[ Better readability] What do you think to create another job (or add this functionality to `pre-flight`) and move all logic there?
in `preflight`:
```
function contains_in() {
   if [[...]]; then return 'true';
   else [[...]]; then return 'false'
}
<...>
echo "RUN_JAX_TEST=$(contains_in base jax ${{ inputs.TEST_SUBSET}} )" >> $GITHUB_OUTPUT
```

in `test-jax`:
```
test-jax:
    needs: build-jax
    if: pre-flight.outputs.RUN_JAX_TEST == 'true'
<...>
```

Reply:
Great suggestion @DwarKapex ! I added it to pre-flight, I like it a little more if the caller decides whether to call a function rather than the callee rejecting its call.

Reply:
Another approach I was think of was to have a dummy jobs with name like `build-jax-4-test`. Then make `test-jax` depends on both `build-jax` and `build-jax-4-test`. The purpose of `build-jax-4-test` is to fail it if you don't want to run `test-jax`. Using dummy job helps to get rid of `if: contains(..)` in test, but seems not an elegant solution